src/v/datalake/coordinator/coordinator.cc

```diff
@@ -47,11 +54,151 @@ std::ostream& operator<<(std::ostream& o, coordinator::errc e) {
     }
 }
 
+void coordinator::start() {
+    ssx::spawn_with_gate(gate_, [this] { return run_until_abort(); });
+}
+
 ss::future<> coordinator::stop_and_wait() {
     as_.request_abort();
+    leader_cond_.broken();
+    if (term_as_.has_value()) {
+        term_as_->get().request_abort();
+    }
     return gate_.close();
 }
 
+ss::future<> coordinator::run_until_abort() {
+    while (!as_.abort_requested()) {
+        auto& raft = *stm_->raft();
+        if (!raft.is_leader()) {
+            bool shutdown = false;
+            auto leader = co_await ss::coroutine::as_future(
+              leader_cond_.wait());
+            if (leader.failed()) {
+                auto eptr = leader.get_exception();
+                if (ssx::is_shutdown_exception(eptr)) {
+                    shutdown = true;
+                }
+            }
+            if (shutdown || as_.abort_requested()) {
+                // Conservative continue.
+                continue;
+            }
+        }
+        auto sync_res = co_await stm_->sync(10s);
+        if (sync_res.has_error()) {
+            auto msg = "Error while syncing coordinator";
+            switch (sync_res.error()) {
+            case coordinator_stm::errc::shutting_down:
+            case coordinator_stm::errc::not_leader:
+                vlog(datalake_log.debug, "{}", msg);
```

Comment:
did you also mean to include the sync_res.error() in the logging? I guess it doesn't matter, we have line numbers in the log line, may be but just for cosmetic reasons.

Reply:
I didn't, but I can add these in a follow up. I think i'd need to implement an ostream operator for it